### PR TITLE
Split edit-period-date into different modes

### PR DESCRIPTION
**Pull request by pmattmann**
Edit Period-Date Dialog.
Split into different modes.
User selects a mode before dialog is opened. Depending on the mode, only some controls are editable.

- move   (moves schedule-entries and day-responsibles)
- add/remove days at start  (schedule-entries and day-responsibles  stay at there date)
- add/remove days at end  (schedule-entries and day-responsibles  stay at there date)

image

## Move
image

## Change start
image

## Change end
image

replaces #3127

### CODE REVIEW

**Review by manuelmeister — APPROVED** [review]

File: frontend/src/components/campAdmin/CampPeriodsListItem.vue [hunk 1]
```diff
@@ -11,65 +11,134 @@ Displays a single period as a list item including controls to edit and delete it
...
+                  $tc('components.campAdmin.campPeriodsListItem.periodChangeStart')
+                }}</v-list-item-title
+              >
+            </v-list-item>
+          </template>
+        </dialog-period-date-edit>
+
+        <dialog-period-date-edit
+          :period="period"
+          modus="changeEnd"
```
Comment (BacLuc) [inline frontend/src/components/campAdmin/CampPeriodsListItem.vue]: I would call that mode in english

**Review by BacLuc — APPROVED** [review]
Thanks, looks good.

A minor issue with denglish

File: frontend/src/components/campAdmin/CampPeriodsListItem.vue (hunk 1 above)
Comment (pmattmann) [reply frontend/src/components/campAdmin/CampPeriodsListItem.vue]: thx

File: frontend/src/plugins/veeValidate/lessThanOrEqual_date.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
+export default (dayjs, i18n) => ({
+  params: ['max'],
+  /**
+   * @param   {string}  value Dater value in local string format
+   * @param   {string}  max   comparison valye in local string format
+   * @returns {bool}          validation result
+   */
+  validate: (value, { max }) => {
```
Comment (usu) [inline frontend/src/plugins/veeValidate/lessThanOrEqual_date.js:8]: A test would be nice, we already have them for `greaterThan` and `greaterThanOrEqual`

File: frontend/src/components/campAdmin/DialogPeriodDescEdit.vue [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    </div>
+  </dialog-form>
+</template>
+
+<script>
+import DialogBase from '@/components/dialog/DialogBase.vue'
+import DialogForm from '@/components/dialog/DialogForm.vue'
+
+export default {
+  name: 'DialogPeriodDescEdit',
```
Comment (usu) [inline frontend/src/components/campAdmin/DialogPeriodDescEdit.vue]: Description? Descending? Descant? I'd vote to avoid any abbr. 😄

**Review by usu — COMMENTED** [review]
Yeah, looks nice & works as intended 👍

**Comment by manuelmeister** [comment]
## Improvements
Add weekday to picker
Add duration as readonly text

**Pull request merged**